lib/rpc/src/rpc.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2021 University of Colorado. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use abomonation::Abomonation;
+use alloc::string::String;
+use alloc::vec::Vec;
+
+#[derive(Debug, Eq, PartialEq, PartialOrd, Clone, Copy)]
+pub enum RPCError {
+    // RPC
+    MissingData,
+    ExtraData,
+    TransportError,
+    MalformedResponse,
+    MalformedRequest,
+    InternalError,
+
+    // File IO
+    InvalidFile,
+    InvalidFlags,
+    InvalidOffset,
+    PermissionError,
+    AlreadyPresent,
+    DirectoryError,
+    OpenFileLimit,
+    FileDescForPidAlreadyAdded,
+    NoFileDescForPid,
+
+    // Syscall Errors
+    InvalidSyscallArgument1 { a: u64 },
+    InvalidVSpaceOperation { a: u64 },
+    InvalidProcessOperation { a: u64 },
+    InvalidSystemOperation { a: u64 },
+
+    // General Errors
+    BadAddress,
+    NotSupported,
+}
+unsafe_abomonate!(RPCError);
+
+#[derive(Debug, Eq, PartialEq, PartialOrd, Clone, Copy)]
+#[repr(u8)]
+pub enum RPCType {
+    /// Client requesting to register with RPC server
+    Registration = 1,
+
+    /// Create a file
+    Create = 2,
+    /// Open a file
+    Open = 3,
+    /// Read from a file
+    Read = 4,
+    /// Read from a file from the given offset
+    ReadAt = 5,
+    /// Write to a file
+    Write = 6,
+    /// Write to a file
+    WriteAt = 7,
+    /// Close an opened file.
+    Close = 8,
+    /// Get the information related to the file.
+    GetInfo = 9,
+    /// Delete the file
+    Delete = 10,
+    /// Write to a file without going into NR.
+    WriteDirect = 11,
+    /// Rename a file.
+    FileRename = 12,
+    /// Create a directory.
+    MkDir = 13,
+
+    Unknown,
+}
+
+pub fn is_fileio(op: RPCType) -> bool {
+    return op >= RPCType::Create && op <= RPCType::MkDir;
```

Comment:
To make this future proof, I'd explicitly list all file variants here (otherwise someone in a few months might come modify RPCType and not realize that it has side effects here